src/app/admin/requests/request-details/request-details.component.ts

```diff
@@ -1,85 +1,179 @@
 import {
   ChangeDetectionStrategy,
+  ChangeDetectorRef,
   Component,
-  OnDestroy,
   OnInit,
 } from '@angular/core';
 import { FormControl, FormGroup, Validators } from '@angular/forms';
 import { SPECIAL_CONDITIONS, ZONES } from '@app/shared/constants';
 import { RequestsFacade } from '../requests.facade';
+import { BeneficiariesService } from '../../beneficiaries/beneficiaries.service';
 import { coordinates } from './request-address-field/request-address-field.component';
-import { RequestTypeUpdated } from '../../../shared/models/requests';
-import { UsersFacade } from '@users/users.facade';
+import { MatDialogRef } from '@angular/material/dialog';
+import { Beneficiary } from '@app/shared/models';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { combineLatest } from 'rxjs';
+import { filter, first } from 'rxjs/operators';
+import { DemandType } from '@app/shared/models/demand';
+import { BeneficiariesFacade } from '@app/admin/beneficiaries/beneficiaries.facade';
 
 @Component({
   templateUrl: './request-details.component.html',
   styleUrls: ['./request-details.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class RequestDetailsComponent implements OnInit, OnDestroy {
+export class RequestDetailsComponent implements OnInit {
   form: FormGroup;
   zones: Array<string> = Object.keys(ZONES).filter((key) => isNaN(+key));
-  needs = RequestTypeUpdated;
+  needs = DemandType;
```

Comment:
Export from file `demand.ts` this, and use it here
```suggestion
  needs = demandTypes;
```
```ts
export const demandTypes = Object.values(DemandType);
```

Reply:
Done